src/lua/LuaGame.cpp

```diff
@@ -93,10 +137,50 @@ static int l_game_start_game(lua_State *l)
  *
  *   experimental
  */
+static void l_game_savegame_stats_callback(std::string filename, Json rootNode)
+{
+	LuaEvent::LuaEvent e("onSaveGameStats");
+	e.addParameter("filename", filename);
+	try {
+		Json gameInfo = rootNode["game_info"];
```

Comment:
Should be `const Json &` here - currently this is a copy-assign which recursively copies the whole node.

Reply:
Yes, my initial version is "dumb" because I don't know C++ very well. Or rather, I don't know how to predict object lifetimes very well, so didn't want things going out of scope and blowing up for now.

I'm old-skool, give me full control over object lifecycles. And yes, I know C++ does that, but it obfuscates it.. which is fine in a small or well-known codebase, but when coming in to a new codebase, until learning in detail all the various object behaviours, it's difficult to understand just when things get constructed and destroyed. I prefer explicit "malloc" and "free".

Reply:
... and reading the rest of your comments, it seems this class should be dropped anyway.

I guess what you're saying is that I can just pass a `lua_State`, or obtain one from within the callback, and then use that to construct the table directly? That was my main sticking point.

Reply:
Yes. The `LuaEvent::Queue()` method (if not passed a specific queue) implicitly operates on the "main" lua_State from `Lua::manager->GetLuaState()` - you can make a Lua table using that state, populate it with the data that needs to be passed back, and then send that table through LuaEvent.

Pushing a `LuaTable` object as a LuaEvent parameter essentially just copies the reference to the underlying Lua table object, which itself has automatically-managed lifetime by the Lua GC.